src/dbnode/storage/index/block.go

```diff
@@ -718,7 +638,32 @@ func (b *block) AggregateWithIter(
 	iter AggregateIterator,
 	opts QueryOptions,
 	results AggregateResults,
-	limit int) error {
+	limit int,
+	logFields []opentracinglog.Field,
+) error {
+	ctx, sp := ctx.StartTraceSpan(tracepoint.BlockAggregate)
+	sp.LogFields(logFields...)
+	defer sp.Finish()
+
+	err := b.aggregateWithSpan(ctx, iter, opts, results, limit)
+	if err != nil {
+		sp.LogFields(opentracinglog.Error(err))
+	}
+	if iter.Done() {
+		b.metrics.aggregateDocsMatched.RecordValue(float64(results.TotalDocsCount()))
+		b.metrics.aggregateSeriesMatched.RecordValue(float64(results.Size()))
```

Comment:
Same here, shouldn't this be done at an end of a query? Not end of querying a block, otherwise the same results will be compoundingly recorded again.